assertions/poster.go

```diff
@@ -140,6 +139,18 @@ func (m *Manager) PostAssertionBasedOnParent(
 	) (protocol.Assertion, error),
 ) (option.Option[*protocol.AssertionCreatedInfo], error) {
 	none := option.None[*protocol.AssertionCreatedInfo]()
+	if m.times.minGapToParent != 0 {
```

Comment:
This would indeed prevent us from posting overflow assertions immediately. The max number of blocks an assertion can consume is 2^26. If there are more than 2^26 blocks between assertions, first, an assertion that consumes 2^26 will be posted, then we should _immediately_ post another assertion based on it that consumes the remaining messages. To know if the parent assertion overflowed, we can check: `protocol.GoGlobalStateFromSolidity(parentCreationInfo.AfterState).PosInBatch > 0`, then we should allow posting its child immediately. We shouldn't wait in that case and should skip this conditional

Reply:
Sorry to bring this up so late. But, I don't think it's sufficient to check that the parentCreationInfo.AfterState PosInBatch > 0.

Imagine that instead of 2^26 the max challenge edge height were just 10.
Imagine you have three batches with 5 messages each when the previous assertion is created.
The previous assertion will say that you need to handle all three batches, but, you'll process the first 10 messages and be at the end of the second block. PosInBatch == 0 but in the 3rd batch. The one you haven't processed yet.
You're still in an overflow assertion because you should be able to immediately post the last 5 messages.

I think the better check is the one that the contracts make:
https://github.com/OffchainLabs/nitro-contracts/blob/bold-merge/src/rollup/RollupCore.sol#L481

Essentially, the machine needs to have reached the expected message from the previous parent assertion.

Reply:
I fixed the check that we already have [check](https://github.com/OffchainLabs/bold/blob/d0a87de774aecfa97161efd1b0a924d4d5fbcf74/assertions/poster.go#L164-L167) to see if an assertion is an overflow assertion or not, and also I'm moving this part of code inside `waitToPostIfNeeded` function as its more appropriate there. 
Thanks!